test/integration/compound/compound_test.go

```diff
@@ -0,0 +1,426 @@
+/*
+ * // SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+ * //
+ * // SPDX-License-Identifier: Apache-2.0
+ */
+
+package compound_test
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/gardener/controller-manager-library/pkg/controllermanager"
+	"github.com/gardener/external-dns-management/pkg/dns"
+	"github.com/gardener/external-dns-management/pkg/dns/provider"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/controller-manager-library/pkg/ctxutil"
+	"github.com/gardener/external-dns-management/pkg/apis/dns/v1alpha1"
+	"github.com/gardener/external-dns-management/pkg/controller/provider/mock"
+)
+
+var _ = Describe("Compound controller tests", func() {
+	const (
+		defaultTTL   = 300
+		retryTimeout = 5 * time.Second
+	)
+
+	var (
+		testRunID      string
+		testNamespace  *corev1.Namespace
+		firstZoneID    dns.ZoneID
+		e1, e2, e3, e4 *v1alpha1.DNSEntry
+
+		checkDeleted = func(g Gomega, ctx context.Context, entry *v1alpha1.DNSEntry) {
+			err := testClient.Get(ctx, client.ObjectKeyFromObject(entry), entry)
+			g.Expect(err).To(HaveOccurred())
+			g.Expect(client.IgnoreNotFound(err)).To(Succeed())
```

Comment:
We should use Gomega's [`ExpectWithOffset`](https://pkg.go.dev/github.com/onsi/gomega#ExpectWithOffset) for all helper functions that perform assertions:
> This is most useful in helper functions that make assertions. If you want Gomega's error message to refer to the calling line in the test (as opposed to the line in the helper function) set the first argument of `ExpectWithOffset` appropriately.
```suggestion
			g.ExpectWithOffset(1, err).To(HaveOccurred())
			g.ExpectWithOffset(1, client.IgnoreNotFound(err)).To(Succeed())
```